### PR TITLE
KTOR-506 Fix selector manager

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-506](https://youtrack.jetbrains.com/issue/KTOR-506)

### CODE REVIEW

**Comment by olme04** [comment]
@e5l can you please describe, where is the main fix happen? Because, while looking on diff, it's hard to understand where the fix is, as there is a lot of renamings :)

**Comment by e5l** [comment]
@olme04, thanks for the notice. Let me check if I pushed all changes

**Comment by e5l** [comment]
@olme04 yep, all is in place. ActorSelectorManager.kt:143 line

**Comment by olme04** [comment]
@e5l thx!

File: ktor-network/jvm/src/io/ktor/network/selector/ActorSelectorManager.kt [hunk 1]
```diff
@@ -30,34 +30,32 @@ public class ActorSelectorManager(context: CoroutineContext) : SelectorManagerSu
...
     init {
         launch {
             val selector = provider.openSelector() ?: error("openSelector() = null")
             selectorRef = selector
-            try {
+            selector.use {
                 try {
-                    process(mb, selector)
-                } catch (t: Throwable) {
+                    process(selectionQueue, it)
```
Comment (rsinukov) [inline ktor-network/jvm/src/io/ktor/network/selector/ActorSelectorManager.kt:43]: please extract `it` to named parameter

**Review by rsinukov — APPROVED** [review]
lgtm

File: ktor-network/jvm/src/io/ktor/network/selector/ActorSelectorManager.kt (hunk 1 above)
Comment (e5l) [reply ktor-network/jvm/src/io/ktor/network/selector/ActorSelectorManager.kt:43]: thanks, sure

**Pull request merged**